sdk/hardwaresecuritymodules/Azure.ResourceManager.HardwareSecurityModules/src/Generated/Models/HardwareSecurityModulesPrivateLinkResource.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System.Collections.Generic;
+using Azure.Core;
+using Azure.ResourceManager.Models;
+
+namespace Azure.ResourceManager.HardwareSecurityModules.Models
+{
+    /// <summary> A private link resource. </summary>
+    public partial class HardwareSecurityModulesPrivateLinkResource : ResourceData
+    {
+        /// <summary> Initializes a new instance of HardwareSecurityModulesPrivateLinkResource. </summary>
+        public HardwareSecurityModulesPrivateLinkResource()
+        {
+            RequiredMembers = new ChangeTrackingList<string>();
+            RequiredZoneNames = new ChangeTrackingList<string>();
+        }
+
+        /// <summary> Initializes a new instance of HardwareSecurityModulesPrivateLinkResource. </summary>
+        /// <param name="id"> The id. </param>
+        /// <param name="name"> The name. </param>
+        /// <param name="resourceType"> The resourceType. </param>
+        /// <param name="systemData"> The systemData. </param>
+        /// <param name="groupId"> The private link resource group id. </param>
+        /// <param name="requiredMembers"> The private link resource required member names. </param>
+        /// <param name="requiredZoneNames"> The private link resource private link DNS zone name. </param>
+        internal HardwareSecurityModulesPrivateLinkResource(ResourceIdentifier id, string name, ResourceType resourceType, SystemData systemData, string groupId, IReadOnlyList<string> requiredMembers, IList<string> requiredZoneNames) : base(id, name, resourceType, systemData)
+        {
+            GroupId = groupId;
+            RequiredMembers = requiredMembers;
+            RequiredZoneNames = requiredZoneNames;
+        }
+
+        /// <summary> The private link resource group id. </summary>
+        public string GroupId { get; }
```

Comment:
What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#d4cebe383870410cab2e14d0709f78bb).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://apiview.dev/Assemblies/Review/25a98e223bda4f39b4ef76bebffa2c23#9f6fc0ae7c8e497e98d7979dadf97a4c).